### ISSUE TITLE
Whitespace change in autogenerated CONFIG docstrings causes Sphinx warnings

### ISSUE THREAD

**Issue by lbianchi-lbl** [IDAES/idaes-pse#1191]
Observed in watertap-org/watertap#1032 as part of the process to update dependencies from IDAES 2.0.0/Pyomo 6.5.0:

![2023-05-25_16-17](https://github.com/IDAES/idaes-pse/assets/48035537/0c24eb86-62c9-4322-9e81-4e56ee9f6441)

The generated warning (one of many, relative to the class shown in the screenshot above) from [this ReadTheDocs build log](https://readthedocs.org/api/v2/build/20821023.txt):

```log
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/1034/watertap/property_models/cryst_prop_pack.py:docstring of watertap.property_models.cryst_prop_pack.NaClParameterBlock:1: WARNING: Block quote ends without a blank line; unexpected unindent.
```

This causes the build to fail if the Sphinx `-W/fail-on-warning` setting is enabled.

**Comment by lbianchi-lbl** [comment]
From a uninformed quick look at the diff between Pyomo 6.5.0 and 6.5.1, this looks like it could be related to changes in `pyomo.common.formatting` and specifically the introduction of `wrap_reStructuredText()`:

https://github.com/IDAES/pyomo/compare/6.5.0...6.5.1.idaes.2023.03.28#diff-1a2731acd796fdfba1065e27dd88538672d8cf61464e62a0a1022cce240c7b80R269

**Comment by andrewlee94** [comment]
@jsiirola Any suggestion on this? Is it something that is in the IDAES usage of the Pyomo documentation functions (e.g. something we need to add to include the trailing linebreak), a difference in the Pyomo tool output, or a bad habit that Pyomo is now being more vocal about?

**Comment by lbianchi-lbl** [comment]
> @jsiirola Any suggestion on this? Is it something that is in the IDAES usage of the Pyomo documentation functions (e.g. something we need to add to include the trailing linebreak), a difference in the Pyomo tool output, or a bad habit that Pyomo is now being more vocal about?

I've just tried using https://pypi.org/project/rstcheck/, and it seems to confirm that the IDAES 2.0.0/Pyomo 6.5.0 output is valid, while the IDAES/idaes-pse@main/Pyomo 6.5.1 is not:

```sh
conda create --yes test-idaes-1191 python=3.10 && conda activate test-idaes-1191
pip install "watertap @ git+https://github.com/watertap-org/watertap@main" rstcheck
pip show pyomo idaes-pse
pip uninstall --yes pyomo idaes-pse && pip install idaes-pse==2.0.0 pyomo=6.5.0 && pip show pyomo idaes-pse
python -c 'from watertap.property_models.cryst_prop_pack import NaClParameterBlock as obj; print(obj.__doc__)' | rstcheck -
# output: Success! No issues detected.
pip uninstall --yes pyomo idaes-pse && pip install "idaes-pse @ git+https://github.com/IDAES/idaes-pse@main" && pip show pyomo idaes-pse
python -c 'from watertap.property_models.cryst_prop_pack import NaClParameterBlock as obj; print(obj.__doc__)' | rstcheck -
# output: <stdin>:17: (WARNING/2) Block quote ends without a blank line; unexpected unindent. Error! Issues detected.
```

**Comment by jsiirola** [comment]
I looked into this and Pyomo/pyomo#2853 restores (most of) the old behavior.  However, relying on the indentation in the ConfigDict declarations to generate correct RST is fragile at best.  This is exactly why Pyomo introduced the use of `wrap_reStructuredText()` in the numpydoc formatter [it also automatically documents types and defaults] (which IDAES is NOT using).

Given that the bulk of IDAES docstrings have settled on google-style doc strings, we should probably look into making a google-style formatter that we use in IDAES.

**Comment by lbianchi-lbl** [comment]
This specific issue (i.e. the backward-incompatible change) has been addressed by Pyomo/pyomo#2853, so this can be closed. #1196 is to discuss more long-term solutions related to the general autogenerated docstring functionality.
